
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://genes-and-health.github.io/TRE-docs/what-genetic-and-health-data-are-where/phenotype-curation/">
      
      
        <link rel="prev" href="../sources-of-raw-health-data/">
      
      
        <link rel="next" href="../../getting-data-in-and-out-of-the-TRE/how-to-get-your-data-in/">
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>Phenotype curation - Genes & Health TRE Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#phenotype-curation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Genes &amp; Health TRE Documentation" class="md-header__button md-logo" aria-label="Genes & Health TRE Documentation" data-md-component="logo">
      
  <img src="../../images/branding/genes-and-health-logo-green.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Genes & Health TRE Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Phenotype curation
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="" data-md-color-scheme="slate" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"/></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/genes-and-health/TRE-docs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Genes &amp; Health TRE Documentation" class="md-nav__button md-logo" aria-label="Genes & Health TRE Documentation" data-md-component="logo">
      
  <img src="../../images/branding/genes-and-health-logo-green.svg" alt="logo">

    </a>
    Genes & Health TRE Documentation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/genes-and-health/TRE-docs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../latest-tre-data-updates-and-changes/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Latest TRE data updates and changes
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Getting started as a new user
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Getting started as a new user
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-started-as-a-new-user/what-is-a-tre/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    What is a TRE?
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-started-as-a-new-user/steps-to-get-access/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Getting access and setting up a profile
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../sandboxes-and-what-each-sandbox-can-access/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Sandboxes and what each sandbox can access
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-started-as-a-new-user/starting-the-tre/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Starting the TRE
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-started-as-a-new-user/shutting-down-the-tre/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Shutting down the TRE
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-started-as-a-new-user/choosing-your-virtual-machine/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Choosing your required virtual machine configuration
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-started-as-a-new-user/how-to-increase-your-disk-size/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    How to increase your disk size
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Using the TRE
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Using the TRE
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../using-the-tre/understanding-tre-folders-and-buckets/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Understanding TRE folders and buckets
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../using-the-tre/handling-files-and-folders/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Handling files and folders
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../using-the-tre/software%2C-modules%2C-requesting-installs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Software, modules, requesting installs
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../using-the-tre/module-manager-guide/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Module Manager User Guide
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../using-the-tre/running-a-gwas/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Running a GWAS
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_4" checked>
        
          
          <label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    What genetic and health data are where?
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            What genetic and health data are where?
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../tre-identifiers/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    TRE identifiers
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../how-are-data-linked/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    How are data linked?
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../sources-of-raw-health-data/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Sources of "raw" health data
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Phenotype curation
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Phenotype curation
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#bi_py-binary-traits" class="md-nav__link">
    <span class="md-ellipsis">
      BI_PY: Binary traits
    </span>
  </a>
  
    <nav class="md-nav" aria-label="BI_PY: Binary traits">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#phenotype-data" class="md-nav__link">
    <span class="md-ellipsis">
      Phenotype data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#input-files" class="md-nav__link">
    <span class="md-ellipsis">
      Input files
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output-files" class="md-nav__link">
    <span class="md-ellipsis">
      Output files
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quant_py-quantitative-traits" class="md-nav__link">
    <span class="md-ellipsis">
      QUANT_PY: Quantitative traits
    </span>
  </a>
  
    <nav class="md-nav" aria-label="QUANT_PY: Quantitative traits">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#phenotype-data_1" class="md-nav__link">
    <span class="md-ellipsis">
      Phenotype data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#input-files_1" class="md-nav__link">
    <span class="md-ellipsis">
      Input files
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Input files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#trait_featurescsv" class="md-nav__link">
    <span class="md-ellipsis">
      trait_features.csv
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trait_aliases_longcsv" class="md-nav__link">
    <span class="md-ellipsis">
      trait_aliases_long.csv
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#unit_conversionscsv" class="md-nav__link">
    <span class="md-ellipsis">
      unit_conversions.csv
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hospital-admission-data" class="md-nav__link">
    <span class="md-ellipsis">
      Hospital admission data
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output-files_1" class="md-nav__link">
    <span class="md-ellipsis">
      Output files
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_5" >
        
          
          <label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Getting data in and out of the TRE
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            Getting data in and out of the TRE
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-data-in-and-out-of-the-TRE/how-to-get-your-data-in/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    How to get your data in
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-data-in-and-out-of-the-TRE/requesting-data-out-of-the-TRE/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Requesting data out of the TRE
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" >
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    How tos
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            How tos
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../how-tos/how-to-update-this-documentation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    How to update this documentation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../how-tos/how-to-use-docker/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    How to use docker
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../how-tos/how-to-write-hash-symbols/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    How to write hash symbols
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-help/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Getting help
    
  </span>
  

      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#bi_py-binary-traits" class="md-nav__link">
    <span class="md-ellipsis">
      BI_PY: Binary traits
    </span>
  </a>
  
    <nav class="md-nav" aria-label="BI_PY: Binary traits">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#phenotype-data" class="md-nav__link">
    <span class="md-ellipsis">
      Phenotype data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#input-files" class="md-nav__link">
    <span class="md-ellipsis">
      Input files
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output-files" class="md-nav__link">
    <span class="md-ellipsis">
      Output files
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#quant_py-quantitative-traits" class="md-nav__link">
    <span class="md-ellipsis">
      QUANT_PY: Quantitative traits
    </span>
  </a>
  
    <nav class="md-nav" aria-label="QUANT_PY: Quantitative traits">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#phenotype-data_1" class="md-nav__link">
    <span class="md-ellipsis">
      Phenotype data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#input-files_1" class="md-nav__link">
    <span class="md-ellipsis">
      Input files
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Input files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#trait_featurescsv" class="md-nav__link">
    <span class="md-ellipsis">
      trait_features.csv
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#trait_aliases_longcsv" class="md-nav__link">
    <span class="md-ellipsis">
      trait_aliases_long.csv
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#unit_conversionscsv" class="md-nav__link">
    <span class="md-ellipsis">
      unit_conversions.csv
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hospital-admission-data" class="md-nav__link">
    <span class="md-ellipsis">
      Hospital admission data
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#output-files_1" class="md-nav__link">
    <span class="md-ellipsis">
      Output files
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
    <a href="https://github.com/genes-and-health/TRE-docs/edit/main/docs/what-genetic-and-health-data-are-where/phenotype-curation.md" title="Edit this page" class="md-content__button md-icon" rel="edit">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M10 20H6V4h7v5h5v3.1l2-2V8l-6-6H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h4zm10.2-7c.1 0 .3.1.4.2l1.3 1.3c.2.2.2.6 0 .8l-1 1-2.1-2.1 1-1c.1-.1.2-.2.4-.2m0 3.9L14.1 23H12v-2.1l6.1-6.1z"/></svg>
    </a>
  
  


<h1 id="phenotype-curation">Phenotype Curation<a class="headerlink" href="#phenotype-curation" title="Permanent link">&para;</a></h1>
<p>The philosophy of the Genes &amp; Health project is to provide researchers with the tools to generate their own phenotypes as well as providing a curated set of phenotypes.  For the latter, the Genes &amp; Health Data Team provides two "pre-computed" phenotypic datasets identified by the name of the python-scripted pipeline used to generate them: <code>BI_PY</code> (binary traits) and <code>QUANT_PY</code> (quantitative traits).</p>
<div class="admonition tip">
<p class="admonition-title">Learning more about <code>BI_PY</code> and <code>QUANT_PY</code></p>
<p>Although <code>BI_PY</code> and <code>QUANT_PY</code> are briefly described below, users wanting further details on these are encouraged to visit the public GitHub pages for each pipeline:</p>
<ul>
<li><a href="https://github.com/genes-and-health/BI_PY/" target="_blank">BI_PY — a python pipeline for binary trait attribution in Genes &amp; Health</a></li>
<li><a href="https://github.com/genes-and-health/QUANT_PY/" target="_blank">QUANT_PY — a python pipeline for quantitative data extraction in Genes &amp; Health</a></li>
</ul>
</div>
<div class="admonition info">
<p class="admonition-title">A note regarding the use of <code>tre-tools</code></p>
<p>The <a href="https://github.com/genes-and-health/tre-tools" target="_blank"><code>tre-tools</code></a> were used until the creation of version 10 of the Genes &amp; Health curated phenotype resources.  Although v10 does use some <a href="https://github.com/genes-and-health/tre-tools" target="_blank"><code>tre-tools</code></a>, changes in the structure and impementation of the phenotype pipielines mean that <a href="https://github.com/genes-and-health/tre-tools" target="_blank"><code>tre-tools</code></a> usage is now limited.  The <a href="https://github.com/genes-and-health/tre-tools" target="_blank"><code>tre-tools</code></a> package is no longer being maintained or updated and is likely to be deprecated in subsequent curated phenotype releases.
<p>
We are very grateful to <a href="https://www.carolinemorton.co.uk/about/" target="_blank">Caroline Morton</a>, <a href="https://uk.linkedin.com/in/saeed-bidi" target="_blank">Saeed Bidi</a> and current and former members of the G&amp;H Data Team for their help in developing the TRE pipeline infrastructure. </p>
</div>
<h2 id="bi_py-binary-traits">BI_PY: Binary traits<a class="headerlink" href="#bi_py-binary-traits" title="Permanent link">&para;</a></h2>
<p>Binary traits are those that have two possible outcomes, such as disease status (e.g. diabetes, hypertension). A person either has the disease or they do not.</p>
<p>The Genes &amp; Health (G&amp;H) <code>BI_PY</code> pipeline extracts and processes binary trait data from G&amp;H phenotype data.  Relevant codes may be SNOMED-CT, ICD-10 or OPCS codes.  <code>BI_PY</code> "runs" a list of such per-binary-phenotype codes against any code associated with a G&amp;H volunteer.  If the volunteer has one or more such codes, they are associated with the relevant binary phenotype --allowing case-control type analyses against volunteers not associated with the binary phenotype.</p>
<p>The pipeline creates files and covariate files suitable for <code>regenie</code> [G/Ex]WAS analysis, as well as generic files for each binary trait at a <em>per individual</em> level (one row per individual summarising the individual’s earliest applicable code and age at first diagnosis).</p>
<p>The pipeline is constituted of 8 sequential python notebooks:</p>
<ul>
<li><strong>1-create-clean-demographics-notebook</strong></li>
<li><strong>2-process-datasets-discovery-primary-care</strong></li>
<li><strong>3-process-datasets-barts-health</strong></li>
<li><strong>4-process-datasets-bradford</strong></li>
<li><strong>5-process-datasets-nhs-digital</strong></li>
<li><strong>6-merge-datasets-notebook</strong></li>
<li><strong>7-three-and-four-digit-ICD</strong></li>
<li><strong>8-custom-phenotypes</strong></li>
</ul>
<h3 id="phenotype-data">Phenotype data<a class="headerlink" href="#phenotype-data" title="Permanent link">&para;</a></h3>
<p>The pipeline imports G&amp;H phenotype data in <code>.../library-red/phenotypes_rawdata/</code>.  These data are from the following sources:</p>
<ol>
<li><strong>DSA__BartHealth_NHS_Trust</strong>: Secondary care data from the Barts Health NHS Trust [North East London: ~40,000 individuals with data]</li>
<li><strong>DSA__BradfordTeachingHospitals_NHSFoundation_Trust</strong>: Secondary care data from the Bradford Teaching Hospitals NHS Trust [Bradford and environs: ~1,700 individuals with data]</li>
<li><strong>DSA__Discovery_7CCGs</strong>: Primary care data from the North East London ICS [North East London: ~45,000 individuals with data]</li>
<li><strong>DSA_NHSDigital</strong>: Data from NHS Digital (NHSD) [England-wide: ~TBC individuals with data].  Data files vary with each cut of NHSD but include one or more of: i) <strong>civil registration data</strong>, ii) <strong>HES APC data</strong>, iii) <strong>HES OP data</strong>, iv) <strong>cancer registry data</strong>, v) <strong>ECDS data</strong></li>
</ol>
<h3 id="input-files">Input files<a class="headerlink" href="#input-files" title="Permanent link">&para;</a></h3>
<p>The pipeline requires a single input file.  This file is a <code>.csv</code> file which gives a set of SNOMED +/- ICD-10 +/- OPCS codes each of which constitute a phenotype.  The current version of this input file is <code>GenesAndHealth_custombinary_codelist_v010_2025_05v4.csv</code>.  This file can be found in the <code>/library-red/genesandhealth/phenotypes_curated/version010_2025_04/BI_PY/inputs/</code> directory.</p>
<p>The custom codelist .csv file has 4 columns: <code>phenotype</code>, <code>code</code>, <code>name</code>, <code>comment</code>.</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-0-1"><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>phenotype, code, name, comment
</span><span id="__span-0-2"><a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>GNH0002_CoronaryArteryDisease_narrow,I200,ICD10,Unstable angina,
</span><span id="__span-0-3"><a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>GNH0002_CoronaryArteryDisease_narrow,I201,ICD10,Angina pectoris with documented spasm,
</span><span id="__span-0-4"><a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>GNH0002_CoronaryArteryDisease_narrow,I208,ICD10,Other forms of angina pectoris,
</span><span id="__span-0-5"><a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>[...]
</span><span id="__span-0-6"><a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>GNH0002_CoronaryArteryDisease_narrow,K401,OPCS4,Saphenous vein graft replacement of one coronary artery,
</span><span id="__span-0-7"><a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>GNH0002_CoronaryArteryDisease_narrow,K402,OPCS4,Saphenous vein graft replacement of two coronary arteries,
</span><span id="__span-0-8"><a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>GNH0002_CoronaryArteryDisease_narrow,K403,OPCS4,Saphenous vein graft replacement of three coronary arteries,
</span><span id="__span-0-9"><a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>[...]
</span><span id="__span-0-10"><a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a>GNH0002_CoronaryArteryDisease_narrow,I753000,SNOMED ConceptID,Old myocardial infarction,
</span><span id="__span-0-11"><a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a>GNH0002_CoronaryArteryDisease_narrow,22298000,SNOMED ConceptID,Heart attack,
</span><span id="__span-0-12"><a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a>GNH0002_CoronaryArteryDisease_narrow,22298000,SNOMED ConceptID,Myocardial infarction,
</span></code></pre></div>
<p>In <code>GenesAndHealth_custombinary_codelist_v010_2025_05v4.csv</code> there are <strong>285 binary trait codelists</strong> from various users:</p>
<ul>
<li><strong>MULTIPLY-initiative</strong>: 202 traits; e.g. Ankylosing_spondylitis [ICD-10, SNOMED-CT, OPCS]</li>
<li><strong>GenomicsPLC/Consortium</strong>: 16 traits; e.g. GNH0005_MyocardialInfarction_extended [ICD-10, SNOMED-CT, OPCS]</li>
<li><strong>Bespoke researcher/research group</strong>: 42 traits; e.g. MGH_MitralValveProlapse [ICD-10 and/or SNOMED-CT and/or OPCS]</li>
<li><strong>NEW! NHS Primary Care Domain refsets</strong>: 25 traits; e.g. QOF_CKD_COD [SNOMED-CT only]</li>
</ul>
<p>The <a href="https://docs.google.com/spreadsheets/d/1ipwdF2j_owfr_QbkDYk1rk0TW3KtdfQYVQn-Vf-o38s/edit?usp=sharing"><code>GenesAndHealth_custombinary_codelist_v010_2025_05v4.csv</code></a> Google document details the binary trait codelists (tab <code>v010_2025_05_BinaryTraits_Custom_Codelists</code>) and give an count of individuals associated with a specific custom or ICD-10 derived codelist. </p>
<h3 id="output-files">Output files<a class="headerlink" href="#output-files" title="Permanent link">&para;</a></h3>
<p>The <code>outputs</code> directory hierarchy is as follows:</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-1-1"><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>outputs  
</span><span id="__span-1-2"><a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>├── custom_phenotypes  
</span><span id="__span-1-3"><a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>│   ├── individual_trait_files  
</span><span id="__span-1-4"><a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>│   └── regenie  
</span><span id="__span-1-5"><a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>└── icd10  
</span><span id="__span-1-6"><a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>    ├── individual_trait_files  
</span><span id="__span-1-7"><a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a>    │   ├── 3_digit_icd  
</span><span id="__span-1-8"><a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a>    │   └── 4_digit_icd  
</span><span id="__span-1-9"><a id="__codelineno-1-9" name="__codelineno-1-9" href="#__codelineno-1-9"></a>    └── regenie
</span></code></pre></div>
<h2 id="quant_py-quantitative-traits">QUANT_PY: Quantitative traits<a class="headerlink" href="#quant_py-quantitative-traits" title="Permanent link">&para;</a></h2>
<p>The Genes &amp; Health (G&amp;H) <code>QUANT_PY</code> pipeline extracts and processes quantitative data from G&amp;H phenotype data.  </p>
<p>The pipeline identifies all primary and secondary care quantitative trait readings; harmonises units; removes duplicates (including serial readings within narrow windows); excludes outlier values above clinically curated reasonable upper and lower bounds; and presents individual and study-level mean, median, min and max values.</p>
<p>It creates files and covariate files suitable for <code>regenie</code> [G/Ex]WAS analysis as well as generic files for each quantitative trait at a <em>per individual</em> level (one row per individual summarising the individual's values for the trait) and a <em>per result</em> level (one line per individual-result).  The pipeline processes HES data to identify admitted patient care (APC) episodes.  From this, three versions of the created files are generated: 1) All data, 2) Out of hospital data (without APC + a buffer), 3) In hospital data (within APC + a  buffer).</p>
<h3 id="phenotype-data_1">Phenotype data<a class="headerlink" href="#phenotype-data_1" title="Permanent link">&para;</a></h3>
<p>The pipeline imports G&amp;H phenotype data in <code>/library-red/phenotypes_rawdata/</code>.  These data are from the following sources:</p>
<ol>
<li><strong>DSA__BartHealth_NHS_Trust</strong>: Secondary care data from the Barts Health NHS Trust [North East London: ~40,000 individuals with data]</li>
<li><strong>DSA__BradfordTeachingHospitals_NHSFoundation_Trust</strong>: Secondary care data from the Bradford Teaching Hospitals NHS Trust [Bradford and environs: ~1,700 individuals with data]</li>
<li><strong>DSA__Discovery_7CCGs</strong>: Primary care data from the North East London ICS [North East London: ~45,000 individuals with data]</li>
<li><strong>DSA_NHSDigital</strong>: Data from the National Diabetes Audit (NDA) [England-wide: ~13,000 individuals with data]</li>
</ol>
<h3 id="input-files_1">Input files<a class="headerlink" href="#input-files_1" title="Permanent link">&para;</a></h3>
<p>The pipeline requires 3 <strong>trait input</strong> files: 1. <code>trait_features.csv</code>, 2. <code>trait_aliases_long.csv</code>, 3. <code>unit_conversions.csv</code>.</p>
<h4 id="trait_featurescsv"><em><code>trait_features.csv</code></em><a class="headerlink" href="#trait_featurescsv" title="Permanent link">&para;</a></h4>
<p>This file lists all the quantitative traits currently extracted from the phenotype data.  The .csv file has 4 columns: trait,target_units,min,max.</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-2-1"><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>trait,target_units,min,max
</span><span id="__span-2-2"><a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>2h postprandial glucose,millimol/L,0.6,45.0
</span><span id="__span-2-3"><a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>AFP,kU/L,1e-100,10000.0
</span><span id="__span-2-4"><a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>ALP,units/L,8.0,1500.0
</span><span id="__span-2-5"><a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>ALT,units/L,5.0,1500.0
</span><span id="__span-2-6"><a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>APTT,seconds,1e-100,100.0
</span><span id="__span-2-7"><a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a>AST,units/L,3.0,1000.0
</span><span id="__span-2-8"><a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a>Albumin,g/L,10.0,80.0
</span><span id="__span-2-9"><a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a>Alcohol units per week,units/week,0.0,350.0
</span></code></pre></div>
<p>So, for example, there is a "2h postprandial glucose" trait which is reported in "millimol/L", excluding any value less than 0.6 millimol/L or over 45.0 millimol/L.</p>
<h4 id="trait_aliases_longcsv"><em><code>trait_aliases_long.csv</code></em><a class="headerlink" href="#trait_aliases_longcsv" title="Permanent link">&para;</a></h4>
<p>SNOMED codes are missing for some of the G&amp;H data pulls, this means that quantitative trait extraction is based on free-text trait descriptions (aka <code>original_term</code> within the script).  This file assigns all valid trait descriptions (aliases) to a trait.</p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-3-1"><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>trait,alias
</span><span id="__span-3-2"><a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>2h_postprandial_glucose,2h postprandial glucose
</span><span id="__span-3-3"><a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>2h_postprandial_glucose,&quot;Glucose tolerance test, 2 hour post prandial (procedure)&quot;
</span><span id="__span-3-4"><a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>...
</span><span id="__span-3-5"><a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>Alcohol_units_per_week,Alcohol units per week
</span><span id="__span-3-6"><a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a>Alcohol_units_per_week,Alcohol units/week (qualifier value)
</span><span id="__span-3-7"><a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a>...
</span><span id="__span-3-8"><a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a>Blood_ketones,Blood ketone level (observable entity)
</span><span id="__span-3-9"><a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a>Blood_ketones,POCT Blood Ketones
</span><span id="__span-3-10"><a id="__codelineno-3-10" name="__codelineno-3-10" href="#__codelineno-3-10"></a>...
</span><span id="__span-3-11"><a id="__codelineno-3-11" name="__codelineno-3-11" href="#__codelineno-3-11"></a>creatinine,Creatinine Serum
</span><span id="__span-3-12"><a id="__codelineno-3-12" name="__codelineno-3-12" href="#__codelineno-3-12"></a>creatinine,Creatinine level (observable entity)
</span></code></pre></div>
<h4 id="unit_conversionscsv"><em><code>unit_conversions.csv</code></em><a class="headerlink" href="#unit_conversionscsv" title="Permanent link">&para;</a></h4>
<p>The same trait may be measured in different units depending of the setting (e.g. primary vs secondary care) or the data source (trust 1 vs trust 2).  This file allows unit conversions if a trait is in a valid but undesired unit which can be converted to a target_unit (as defined in <code>trait_features.csv</code>).  It also acts as a synonym dictionary to standardise unit terminology, for example, <code>nmol/L</code> is converted into the preferred term <code>nanomol/L</code>. Such conversions can be identified by a <code>multiplication_factor</code> of 1.0. </p>
<div class="language-text highlight"><pre><span></span><code><span id="__span-4-1"><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>result_value_units,target,multiplication_factor
</span><span id="__span-4-2"><a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>%,%,1.0
</span><span id="__span-4-3"><a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>*10^9/l,10^9/L,1.0
</span><span id="__span-4-4"><a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a>g/L,mg/L,1000.0
</span><span id="__span-4-5"><a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>IU/L,units/L,1.0
</span><span id="__span-4-6"><a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>Kg,grams,1000.0
</span><span id="__span-4-7"><a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a>mg/L,g/L,0.001
</span><span id="__span-4-8"><a id="__codelineno-4-8" name="__codelineno-4-8" href="#__codelineno-4-8"></a>miu/L,milliunits/L,1.0
</span><span id="__span-4-9"><a id="__codelineno-4-9" name="__codelineno-4-9" href="#__codelineno-4-9"></a>nmol/L,nanomol/L,1.0
</span><span id="__span-4-10"><a id="__codelineno-4-10" name="__codelineno-4-10" href="#__codelineno-4-10"></a>Units/Day,units/week,7.0
</span></code></pre></div>
<h4 id="hospital-admission-data">Hospital admission data<a class="headerlink" href="#hospital-admission-data" title="Permanent link">&para;</a></h4>
<p><code>QUANT_PY</code> uses NHS England Digital Hospital Episode Statistics (HES) Admitted Patient Care (APC) data to identify periods of hospitalisation.  As certain hospitals log day treatments as APC events (e.g. immunotherapy infusions), <strong>only APC episodes &gt;2 calendar days are considered as hospitalisation</strong>.  At present, <code>QUANT_PY</code> does not exclude data obtained during a A&amp;E episode (HES AE + ECDS) unless this leads to a hospital admission (in which case it is "subsumed" by an APC episode).  However, the script is written such that it could accommodate these if needed/desired.</p>
<p><code>QUANT_PY</code> <strong>extends the hospitalisation episode by a 2-week buffer on either side of the APC event</strong> on the basis that individuals admitted to hospital are typically unwell in the days leading to hospitalisation and may be discharged recovering, but prior to a return to their baseline status.</p>
<h3 id="output-files_1">Output files<a class="headerlink" href="#output-files_1" title="Permanent link">&para;</a></h3>
<p>The following files are generated by <code>QUANT_PY</code> to the  <code>/library-red/genesandhealth/phenotypes_curated/version010_2025_04/QUANT_PY/outputs/</code> directory:</p>
<ol>
<li><strong>per trait files</strong> [<code>../outputs/individual_trait_files/</code>; subdirectories: <code>in_hospital</code>, <code>out_hospital</code>, <code>all</code>]:<ul>
<li><strong><code>_{trait}_readings_at_unique_timepoints.csv</code></strong>: one validated result per row (columns: <code>pseudo_nhs_number, trait, unit, value, date, gender, age_at_test, minmax_outlier</code>) </li>
<li><strong><code>_{trait}_per_individual_stats.csv</code></strong>: one row per volunteer (<code>pseudo_nhs_number, trait, median, mean, max, min, earliest, latest, number_observations</code>)</li>
</ul>
</li>
<li><strong>per trait plots</strong> [<code>../outputs/individual_trait_plots/</code>; subdirectories: <code>in_hospital</code>, <code>out_hospital</code>, <code>all</code>]:<ul>
<li><strong><code>_{trait}_{setting}.svg</code></strong>: Histograms of trait log10(values) for trait separated M and F listing median, mean, min, max, number individuals, number observations</li>
</ul>
</li>
<li><strong>regenie files</strong> [<code>../outputs/regenie/</code>; subdirectories: <code>in_hospital</code>, <code>out_hospital</code>, <code>all</code> and <code>covariate_files</code>]:<ul>
<li><strong><code>_{trait}_{setting}_[regenie_51|regenie_55].tsv</code></strong>: regenie files for 51kGWAS and 55kExome analyses</li>
<li><strong><code>./covariate_files/_{setting}_[regenie_51|regenie_55]_megawide.tsv</code></strong>: regenie covariate files allowing age at test analyses (cf. age on joining Genes and Health)</li>
</ul>
</li>
</ol>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../sources-of-raw-health-data/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Sources of &#34;raw&#34; health data">
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Sources of "raw" health data
              </div>
            </div>
          </a>
        
        
          
          <a href="../../getting-data-in-and-out-of-the-TRE/how-to-get-your-data-in/" class="md-footer__link md-footer__link--next" aria-label="Next: How to get your data in">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                How to get your data in
              </div>
            </div>
            <div class="md-footer__button md-icon">
              
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "../..", "features": ["content.code.copy", "content.code.select", "content.code.annotate", "content.action.edit", "navigation.footer", "navigation.path", "navigation.instant", "navigation.tracking"], "search": "../../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
        <script src="../../js/edit-link-newtab.js"></script>
      
    
  </body>
</html>