# Latest TRE data updates and changes {#latest-tre-data-updates-and-changes}

(newest shown first)

## 2025-MAY-20 by DVH  
>    Begun release of 55k exome ExWAS to Consortium1 partners. Quant traits initially, to be followed by binary traits.  
gs://qmul-production-library-consortiumpriorityperiod-red/Callsets/2024\_05\_55kCallset/release\_2025-JUN-16\_ExWAS/

## 2025-MAY-05 by DVH  
>    Released 51k GSA-TOPMEDr3 GWAS results for v010 quant traits  
gs://qmul-production-library-red/genesandhealth/GSAv3EAMD/Jan2024\_51k\_TOPMED-r3\_Imputation\_b38/2025\_05\_v010\_quantpy\_GSATOPMEDr3\_GWAS/

## 2025-APR-28 by RK  
>    Improved the file dialog that allows users to upload files to the red bucket. Users can now upload files to the red bucket in the directory of their choice and create a new directory in the red bucket if they wish.

## 2025-APR-24 by DVH  
>    More Barts Health datafiles from 2024\_09 release redaacted and put in the TRE  
gs://qmul-production-library-red/genesandhealth/phenotypes\_rawdata/DSA\_\_BartsHealth\_NHS\_Trust/2024\_09\_ResearchDataset/

## 2025-APR-24 by DVH  
>    New quant traits v010 uploaded,   
gs://qmul-production-library-red/genesandhealth/phenotypes\_curated/version010\_2025\_04/

## 2025-MAR-26 by DVH  
>    Uploaded new NHS England (ex Digital) HES data. More datasets still to come.  
gs://qmul-production-library-vnhsdigital-sublicence-red/DSA\_\_NHSDigitalNHSEngland/2025\_03/

## 2025-MAR-26 by DVH  
>    Uploaded new folder of gvcf files from exome sequencing. These are the input to variant joint calling. They are unfiltered samples and genotypes, do not use for genetic association studies\!  
gs://qmul-production-library-consortiumpriorityperiod-red/gvcfs/

## 2025-MAR-25 by DVH  
>    Removal of some files in previous Barts Health phenotypes\_raw datasets, and replacement with more highly redacted versions.

## Deleted  
>    `gs://qmul-production-library-red/genesandhealth/phenotypes\_rawdata/DSA\_\_BartsHealth\_NHS\_Trust/2024\_09\_ResearchDataset/RDE\_Measurements.ascii.redacted.tab`  
>    **Replaced with:**    
>   `gs://qmul-production-library-red/genesandhealth/phenotypes\_rawdata/DSA\_\_BartsHealth\_NHS\_Trust/2024\_09\_ResearchDataset/RDE\_Measurements.ascii.redacted2.tab`

## Deleted:  
>    `gs://qmul-production-library-red/genesandhealth/phenotypes\_rawdata/DSA\_\_BartsHealth\_NHS\_Trust/2023\_12\_ResearchDatasetv1.6/GandH\_Measurements\_\_20240423.ascii.redacted.tab`
>    **Replaced with:**
>    `gs://qmul-production-library-red/genesandhealth/phenotypes\_rawdata/DSA\_\_BartsHealth\_NHS\_Trust/2023\_12\_ResearchDatasetv1.6/GandH\_Measurements\_\_20240423.ascii.redacted2.tab`

## Deleted \- we do not think this file is used at all, so have not replaced.  
>    `gs://qmul-production-library-red/genesandhealth/phenotypes\_rawdata/DSA\_\_BartsHealth\_NHS\_Trust/2023\_12\_ResearchDatasetv1.6/in-beta-testing/GH\_RAW\_Pathology\_\_20231211.ascii.redacted.tab`

## Deleted:  
>    `gs://qmul-production-library-red/genesandhealth/phenotypes\_rawdata/DSA\_\_BartsHealth\_NHS\_Trust/2024\_09\_ResearchDataset/RDE\_Pathology.ascii.nohisto.redacted.csv`
>    **Replaced with:**
>    `gs://qmul-production-library-red/genesandhealth/phenotypes\_rawdata/DSA\_\_BartsHealth\_NHS\_Trust/2024\_09\_ResearchDataset/RDE\_Pathology.ascii.nohisto.redacted2.csv`

## Deleted:  
>    `gs://qmul-production-library-red/genesandhealth/phenotypes\_rawdata/DSA\_\_BartsHealth\_NHS\_Trust/2023\_12\_ResearchDatasetv1.6/GH\_Pathology\_\_20231218.ascii.nohisto.redacted.tab`
>    **Replaced with: **
>    `gs://qmul-production-library-red/genesandhealth/phenotypes\_rawdata/DSA\_\_BartsHealth\_NHS\_Trust/2023\_12\_ResearchDatasetv1.6/GH\_Pathology\_\_20231218.ascii.nohisto.redacted2.tab`
